python/python/lance/_dataset/sharded_dataset.py

```diff
@@ -0,0 +1,143 @@
+#  Copyright (c) 2024. Lance Developers
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from pathlib import Path
+from typing import Generator, List, Literal, Optional, Union
+
+import pyarrow as pa
+
+import lance
+from lance.dataset import LanceDataset
+
+__all__ = ["ShardedDataset"]
+
+
+def _sharded_fragments(
```

Comment:
would it make sense to make these methods so you don't have to pass in as many parameters?